libraries/exoplayer_hls/src/main/java/androidx/media3/exoplayer/hls/HlsMediaSource.java

```diff
@@ -557,7 +559,7 @@ private SinglePeriodTimeline createTimelineForLive(
     return new SinglePeriodTimeline(
         presentationStartTimeMs,
         windowStartTimeMs,
-        /* elapsedRealtimeEpochOffsetMs= */ C.TIME_UNSET,
+        SntpClient.getElapsedRealtimeOffsetMs(),
         periodDurationUs,
```

Comment:
If the SntpClient is not initialized this change done nothing, so you could consider making the changes to `DefaultHlsPlaylistTracker` surrounded by a feature flag from the `HlsPlaylistTracker.Factory` (if it was not private), or simply a PSFB flag.